### PR TITLE
Added cli_serializer & from_cli function

### PR DESCRIPTION
**Pull request by Denperidge**
---
name: Pull request
about: Submit a pull request for this project
assignees: fabiocaccamo

---

**Describe your changes**
- Implemented a from_cli function and its respective serializer + tests

**Related issue/discussion**
https://github.com/fabiocaccamo/python-benedict/discussions/290

**Extra notes:**
- ArgumentParser allows for an auto-generated --help interface, but this not possible in the current implementation. Support for this could be added, but it would require extra top level functions, and I am unsure as to whether that would be overkill or out of scope
- It should be possible to allow from_cli to take 0 arguments, as the internally used parse_args function also allows 0-1 arguments (in case of 0, it takes the arguments passed to the currently running script). However, I don't know if that behaviour is desirable, as it would make from_cli more unlike the other from_* functions
- Technically, to_cli could also be possible (e.g. create a string with the command that would need to be run to get these options), but I don't know if that has much use

**Checklist before requesting a review**
- [x] I have performed a self-review of my code.
- [x] I have added tests for the proposed changes.
- [x] I have run the tests and there are not errors.

### CODE REVIEW

**Comment by fabiocaccamo** [comment]
@Denperidge thank you very much for this PR!
I see there are some errors on Python 3.8, could you fix them?
I will review this as soon as possible.

**Comment by Denperidge** [comment]
My pleasure!

Got the error! The exit_on_error option got added in 3.9. But because of the error catching that is done further down, it should be redundant either way.

Take your time to review, and have a nice day!

File: benedict/serializers/cli.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+from re import finditer
+
+from benedict.serializers.abstract import AbstractSerializer
+from benedict.utils import type_util
+
+regex_keys_with_values = r"-+\w+(?=\s[^\s-])"
+regex_all_keys = r"-+\w+"
+
+
+def parse_keys(regex, string):
```
Comment (fabiocaccamo) [inline benedict/serializers/cli.py:12]: Maybe this could be a static method of `CLISerializer`

File: benedict/serializers/cli.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                # and is defined once, store as bool
+                if count <= 1:
+                    parser.add_argument(key, action="store_true", required=False)
+                # and is defined multiple times, count how many times
+                else:
+                    parser.add_argument(key, action="count", required=False)
+
+        try:
+            args = parser.parse_args(s.split())
+        except ArgumentError as error:
```
Comment (fabiocaccamo) [inline benedict/serializers/cli.py:68]: Could you add a test case that raises this exception?

**Review by fabiocaccamo — CHANGES_REQUESTED** [review]

File: benedict/serializers/cli.py (hunk 2 above)
Comment (Denperidge) [reply benedict/serializers/cli.py:68]: On it!

File: benedict/serializers/cli.py (hunk 1 above)
Comment (Denperidge) [reply benedict/serializers/cli.py:12]: Yeah that'd make more sense! I also added the regex_ strings into the class, but if you prefer they *do* remain outside feel free to (ask me to) change it

**Review by fabiocaccamo — APPROVED** [review]

**Comment by fabiocaccamo** [comment]
@Denperidge release within `0.32.0` version.

**Comment by Denperidge** [comment]
Amazing, thanks for helping me integrate it! @fabiocaccamo Have a nice day ♥

**Comment by fabiocaccamo** [comment]
Thank you for your great contribution! H
ave a nice day you too!

**Pull request merged**